src/lib/render-lit-html.ts

```diff
@@ -566,19 +594,34 @@ export function* renderTemplateResult(
         // Instantiate the element and stream its render() result
         let instance = undefined;
         try {
-          const element = new ctor();
-          (element as any).tagName = op.tagName;
-          instance = new LitElementRenderer(element);
+          instance = new ctor();
+          (instance as any).tagName = op.tagName;
+          for (const [name, value] of op.staticAttributes) {
+            instance.setAttribute(name, value);
+          }
         } catch (e) {
-          console.error('Exception in custom element constructor', e);
+          console.error(`Exception in custom element callback for ${op.tagName}`, e);
         }
         renderInfo.customElementInstanceStack.push(instance);
         break;
       }
-      case 'custom-element-render': {
+      case 'custom-element-render-attributes': {
         const instance = getLast(renderInfo.customElementInstanceStack);
         if (instance !== undefined) {
-          yield* instance.renderElement();
+          if (instance.connectedCallback) {
+            instance?.connectedCallback();
```

Comment:
We still need to more strictly define the server-side lifecycle, but right now I feel strange about calling connectedCallback on the server. That seems like it could be a good signal that the DOM is available. Can we call the methods that it calls directly?